README.md

```diff
@@ -15,106 +15,38 @@ agents with Hashicorp Vault.
 #### Table of Contents
 
 1. [Description](#description)
-2. [Setup - The basics of getting started with vault_lookup](#setup)
-3. [Usage - Configuration options and additional functionality](#usage)
+2. [Requirements](#setup)
+3. [Usage, Configuration, and Examples](#usage)
+4. [Authentication Methods](#authentication-methods)
 
 ## Description
 
-For users with a Puppet Enterprise 2019 or open source Puppet 6 infrastructure
-wanting to leverage secrets from an existing [Hashicorp
-Vault](https://www.vaultproject.io/) server. Used with Puppet 6's Deferred type,
-this allows agents to retrieve secrets from Vault when a catalog is applied. In
-this way, the secret data is not embedded in the catalog and the master never
-sees it. See [this blog
+For Puppet 6+ or Puppet Enterprise 2019+ users wanting to use secrets from
+[Hashicorp Vault](https://www.vaultproject.io/) on their Puppet agents, this
+Puppet module provides the `vault_lookup::lookup()` function.
+
+When used with Puppet 6's [`Deferred`
+type](https://puppet.com/docs/puppet/7/deferring_functions.html), the function
+allows agents to retrieve secrets from Vault when a catalog is applied rather
+than compiled. In this way, the secret data is not embedded in the catalog and
+the Puppetserver does not need permissions to read all your Vault secrets.
+
+See [this blog
 post](https://puppet.com/blog/secret-agents-man-secrets-store-integrations-puppet-6)
```

Comment:
Broken link :disappointed: 

Available on the internet archive:
https://web.archive.org/web/20220520045735/https://www.puppet.com/blog/secret-agents-man-secrets-store-integrations-puppet-6/

Reply:
Aww, bummer.
Are we okay with linking to the web archive?

Reply:
Not sure it's the best: to me it looks like promotion of Puppet 6 which was just announced, and now that Puppet 6 is EOL, it has low value maybe?  The current documentation should have all the content in a more up-to-date form I believe :shrug: .

Reply:
Yup, agreed that that blog post isn't really useful.
I'll either drop this line entirely, or find something more useful to link to.

Reply:
if you cannot find an alternative I think the link is fine. also CC @binford2k about the broken link

Reply:
Decided to just drop the link.